server/src/Service/InteractionService.php

```diff
@@ -32,8 +40,22 @@ public function getInteraction(
 		$this->validateUsers( $users );
 		$this->validateLimit( $limit );
 
+		// get user ids from usernames
+		foreach ( $users as $username ) {
+			$userId = $this->userDao->getUserId( $username );
+			if ( !$userId ) {
+				throw new \InvalidArgumentException( sprintf( 'username %s could not be found', $username ) );
+			}
+			$userIds[] = $userId;
+		}
+
+		// get sensible starting date if no date is provided
+		if ( !$startDate && !$endDate ) {
```

Comment:
This should probably execute even if I send an `endDate`

Reply:
I thought about it, but then it could happen that $endDate is before the "calculated" $startDate and it will show no results. If that's ok I'll just check for $startDate then

Reply:
uhh.... maybe you should add a validation after the start date is generated and throw an error if that is the case?